src/wolf.system/CMakeLists.txt

```diff
@@ -10,51 +9,68 @@ set(LIBRARY_TYPE "SHARED" CACHE STRING "Library type")
 
 if("${TARGET}" STREQUAL "IOS")
     set(IOS TRUE)
-endif()													
+    set(CMAKE_OSX_SYSROOT iphoneos)
+    set(CMAKE_SYSTEM_NAME iOS)
+    set(CMAKE_OSX_ARCHITECTURES ${ARCH})
+    # signing the project
+    #set_xcode_property(${PROJECT_NAME} CODE_SIGN_IDENTITY "Apple Development")  
+    #set_xcode_property(${PROJECT_NAME} DEVELOPMENT_TEAM ${DEVELOPMENT_TEAM_ID})
+endif()
+
 if(UNIX AND NOT APPLE AND NOT IOS)
    set(LINUX TRUE)
 endif()
 
 # set the project name
 if (IOS)
-    project (wolf_system_ios)						 
+    project(wolf_system_ios VERSION 2.1.0)
 elseif (APPLE)
-	project (wolf_system_osx)
+    project(wolf_system_osx VERSION 2.1.0)
 elseif (WIN32)
-	project (wolf_system_win)
+    project(wolf_system_win VERSION 2.1.0)
 elseif (LINUX)
-	project (wolf_system_linux)
+    project(wolf_system_linux VERSION 2.1.0)
 endif()
 
-if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-    if (WIN32)
-        set(WIN64 TRUE)
-    endif()
+add_library(${PROJECT_NAME} ${LIBRARY_TYPE})
+add_library(wolf::system ALIAS ${PROJECT_NAME})
+
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
```

Comment:
Please do not repeat the same options, properties and etc multiple times. It increases the chance of mistakes in updating those options (updating some parts may be forgotten unintentionally). 

For example, the c++ standard is set to cxx-20 here and in the following by updating `CMAKE_CXX_STANDARD` at the same time.

Reply:
there are three options to resolve this:
1. `target_compiletime_features(${PROJECT_NAME} "cxx_std_${CMAKE_CXX_STANDARD}")`
2. option 1 and/or keeping those two lines close to each other.
3. removing `CMAKE_CX_STANDARD`
4. removing `target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)`

I'd recommend either 1 or 2.

which do you want me to apply?

Reply:
I went with 2, using approach 1; I'll change it if you disagree.